### PR TITLE
Make test running concurrency dynamic based on nprocs

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

In #1228 we decreased the concurrency used for the test runner in CI to
avoid memory pressure and failing jobs. In that commit we set it to 2 as
a quick fix so we could unblock CI and continue landing patches. A
followup was to do the analysis and find the sweet spot for concurrency
and stability. At the same time we opened mtreinish/stestr#202 to add
support to the test runner for running fractional nprocs. That is still
under discussion, but in the meantime this commit adds support to the
makefile for using fractional nprocs concurrency. It adjusts are default
concurrency from 2 to be `nprocs / 2` if nprocs > 3. This should enable us
to increase the number of test workers we run on travis linux jobs but
still maintain reliability.

### Details and comments

### CODE REVIEW

**Comment by mtreinish** [comment]
Hmm, it looks like even with this it's still running on 2 cores. I'm wondering if the types of VMs we're getting from travis are different now and they only have 4 cores. Maybe this isn't needed anymore and we can just go back to not specifying a concurrency and just using nprocs.

**Comment by Nugaziqua** [comment]
Ive tried to unsubscribe numerous times and it wont work. Please someone
get me off the thread...

File: Makefile [hunk 1]
```diff
@@ -3,6 +3,28 @@
...
+ifeq ($(NPROCS), 2)
+	CONCURRENCY := 2
+else ifeq ($(NPROCS), 1)
+	CONCURRENCY := 1
+else ifeq ($(NPROCS), 3)
+	CONCURRENCY := 3
+else ifeq ($(NPROCS), 0)
+	CONCURRENCY := 0
+else
+	CONCURRENCY := $(shell echo "$(NPROCS) 2" | awk '{print $$1 / $$2}')
```
Comment (kdk) [inline Makefile]: We should make sure to get an integer back from awk here.

Comment (mtreinish) [reply Makefile]: Done using printf in: https://github.com/Qiskit/qiskit-terra/pull/1973/commits/12918d5cefd51356ead703c59af10128c7ab5b97

**Pull request merged**